packages/truffle-codec-utils/src/types/types.ts

```diff
@@ -18,33 +18,40 @@ const debug = debugModule("decode-utils:types:types");
 //Similarly with global structs and enums.
 //Foreign contract types aren't really implemented yet either, although
 //those aren't produced from definitions.
+//(General external functions aren't really implemented yet either for the
```

Comment:
It means ones with `kind = "general"`, i.e., ones with no more specific type than "this is an external function".  We need this possbility for handling decoding via ABI (since that's all the ABI has to say about parameters of function type), but decoding via ABI isn't implemented yet, only decoding via definition.  And if we have the definition, well, we know the full type.